2023-10-07 17:15:20,832 INFO [streaming_server.py:812] {'encoder': './sherpa-onnx-streaming-zipformer-en-2023-06-21/encoder-epoch-99-avg-1.int8.onnx', 'decoder': './sherpa-onnx-streaming-zipformer-en-2023-06-21/decoder-epoch-99-avg-1.int8.onnx', 'joiner': './sherpa-onnx-streaming-zipformer-en-2023-06-21/joiner-epoch-99-avg-1.int8.onnx', 'paraformer_encoder': None, 'paraformer_decoder': None, 'tokens': './sherpa-onnx-streaming-zipformer-en-2023-06-21/tokens.txt', 'sample_rate': 16000, 'feat_dim': 80, 'provider': 'cpu', 'decoding_method': 'greedy_search', 'num_active_paths': 4, 'use_endpoint': 1, 'rule1_min_trailing_silence': 2.4, 'rule2_min_trailing_silence': 1.2, 'rule3_min_utterance_length': 20, 'hotwords_file': '', 'hotwords_score': 1.5, 'port': 6006, 'nn_pool_size': 1, 'max_batch_size': 50, 'max_wait_ms': 10, 'max_message_size': 1048576, 'max_queue_size': 32, 'max_active_connections': 500, 'num_threads': 2, 'certificate': None, 'doc_root': 'web'}
2023-10-07 17:15:23,134 INFO [streaming_server.py:626] No certificate provided
2023-10-07 17:15:23,136 INFO [server.py:711] server listening on [::]:6006
2023-10-07 17:15:23,137 INFO [server.py:711] server listening on 0.0.0.0:6006
2023-10-07 17:15:23,138 INFO [streaming_server.py:652] Please visit one of the following addresses:

  http://localhost:6006

Since you are not providing a certificate, you cannot use your microphone from within the browser using public IP addresses. Only localhost can be used.You also cannot use 0.0.0.0 or 127.0.0.1
2023-10-07 17:15:33,669 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,670 INFO [server.py:268] connection closed
2023-10-07 17:15:33,726 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,726 INFO [server.py:268] connection closed
2023-10-07 17:15:33,728 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,729 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,729 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,729 INFO [server.py:268] connection closed
2023-10-07 17:15:33,729 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,729 INFO [server.py:268] connection closed
2023-10-07 17:15:33,730 INFO [server.py:268] connection closed
2023-10-07 17:15:33,730 INFO [server.py:268] connection closed
2023-10-07 17:15:33,860 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,860 INFO [server.py:268] connection closed
2023-10-07 17:15:33,974 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,975 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,976 INFO [server.py:233] connection failed (200 OK)
2023-10-07 17:15:33,976 INFO [server.py:268] connection closed
2023-10-07 17:15:33,976 INFO [server.py:268] connection closed
2023-10-07 17:15:33,976 INFO [server.py:268] connection closed
2023-10-07 17:15:34,662 INFO [server.py:646] connection open
2023-10-07 17:15:34,662 INFO [streaming_server.py:693] Connected: ('::1', 49823, 0, 0). Number of connections: 1/500
2023-10-07 17:15:41,007 INFO [streaming_server.py:677] Disconnected: ('::1', 49823, 0, 0). Number of connections: 0/500
2023-10-07 17:15:41,008 ERROR [server.py:242] connection handler failed
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/websockets/legacy/server.py", line 240, in handler
    await self.ws_handler(self)
  File "/Users/fawazahamedshaik/github/ASR/Interactive_ASR/./streaming_server.py", line 670, in handle_connection
    await self.handle_connection_impl(socket)
  File "/Users/fawazahamedshaik/github/ASR/Interactive_ASR/./streaming_server.py", line 728, in handle_connection_impl
    n = text[0][0:20]
        ~~~~^^^
IndexError: string index out of range
2023-10-07 17:15:41,012 INFO [server.py:268] connection closed
